tasks/model.go

```diff
@@ -24,90 +23,39 @@ type NodeConfig struct {
-	// Stage is a more detailed identifier for what part of the process the deal is in. Some stages are unique to the task type
-	// When a task status is Successful or Failure, Stage indicates the final stage of the deal making process that was reached
-	Stage string `json:"stage"`
-	// CurrentStageDeatails offers more information about what is happening in the current stage of the deal making process
-	CurrentStageDetails *StageDetails `json:"current_stage_details,omitempty"`
-	// StartedAt the time the task was assigned first assigned to a dealbot
-	StartedAt time.Time `json:"started_at,omitempty"`
-	// RetrievalTask is subparameters for a retrieval deal -- will be nil for a storage deal
-	RetrievalTask *RetrievalTask `json:"retrieval_task,omitempty"`
-	// StorageTask is subparameters for a storage deal -- will be nil for a retrieval deal
-	StorageTask *StorageTask `json:"storage_task,omitempty"`
-	// Signature is the crytographic signature of all the data in this task absent the signature field itself (i.e. sig set to nil)
-	Signature []byte `json:"signature,omitempty"`
-}
-
-// StageDetails offers detailed information about progress within the current stage
-type StageDetails struct {
-	// Human-readable fields.
-	Description      string    `json:"description,omitempty"`
-	ExpectedDuration string    `json:"expected_duration,omitempty"`
-	Logs             []*Log    `json:"logs"`
-	UpdatedAt        time.Time `json:"updated_at,omitempty"`
-}
-
-// Log is a message about something that happened in the current stage
-type Log struct {
-	Log       string    `json:"log"`
-	UpdatedAt time.Time `json:"updated_at,omitempty"`
-}
-
 // TaskEvent logs a change in either status
 type TaskEvent struct {
 	Status Status
 	Stage  string
 	At     time.Time
 }
 
-func (t Task) Bytes() []byte {
-	b, err := json.Marshal(&t)
-	if err != nil {
-		return []byte{}
-	}
-	return b
-}
-
-func (t *Task) Sign(privKey crypto.PrivKey) error {
-	var err error
-	t.Signature = nil
-	t.Signature, err = privKey.Sign(t.Bytes())
-	return err
-}
-
-func (t *Task) Log(log *logging.ZapEventLogger) {
-	if t.RetrievalTask != nil {
-		log.Infow("retrieval task", "uuid", t.UUID, "status", t.Status, "worked_by", t.WorkedBy)
-	} else if t.StorageTask != nil {
-		log.Infow("storage task", "uuid", t.UUID, "status", t.Status, "worked_by", t.WorkedBy)
-	} else {
-		panic("both tasks are nil")
-	}
-}
-
-// Status is a global task status, shared among all tasks types
-type Status int
-
-const (
+var (
 	// Available indicates a task is ready to be assigned to a deal bot
-	Available Status = iota + 1
+	Available Status = &_Status{x: 1}
 	// InProgress means the task is running
-	InProgress
+	InProgress Status = &_Status{x: 2}
 	// Successful means the task completed successfully
-	Successful
+	Successful Status = &_Status{x: 3}
 	// Failed means the task has failed
-	Failed
+	Failed Status = &_Status{x: 4}
 )
 
+func (f *_Status__Prototype) Of(x int) Status {
```

Comment:
I'd make this a standalone func, like `StatusOf`, because this assumes a little too much about the codegen. You also don't need it to be a method, I think.

An advantage of not relying on codegen details is that we could later move to node reflection without having to rewrite this func, for example :)